jax/_src/compiler.py

```diff
@@ -314,23 +315,69 @@ def compile_or_get_cached(
     # storage.
     # TODO: In case when coordinator process is not participating in the
     # computation we need to choose another host to compile the module.
-    if distributed.global_state.service is None:
-      serialized_executable = global_client.blocking_key_value_get_bytes(
-          module_name, share_timeout)
-      serialized_executable = compilation_cache.decompress_executable(
-          serialized_executable)
-      executable = backend.deserialize_executable(
-          serialized_executable, compile_options)
-    else:
+    if distributed.global_state.process_id == 0:
       executable = backend_compile(backend, computation,
                                    compile_options, host_callbacks)
       serialized_executable = backend.serialize_executable(executable)
       serialized_executable = compilation_cache.compress_executable(
           serialized_executable)
       global_client.key_value_set(module_name, serialized_executable)
-
+    else:
+      serialized_executable = global_client.blocking_key_value_get_bytes(
+          module_name, share_timeout)
+      serialized_executable = compilation_cache.decompress_executable(
+          serialized_executable)
+      executable = backend.deserialize_executable(
+          serialized_executable, compile_options)
     compile_or_get_cached.cached_modules[module_name] = executable
     return executable
+  elif (
+      process_count() > 1
+      and config.share_autotune_config_between_hosts.value
+      and distributed.global_state.client is not None
+  ):
+    share_timeout = config.share_binary_between_hosts_timeout_ms.value
+    global_client = distributed.global_state.client
+    debug_options = compile_options.executable_build_options.debug_options
+    autotune_tmp_file = os.path.join(
+        compile_or_get_cached.autotune_configs_dir, module_name
+    )
+
+    # If module were previously compiled use existing config.
+    if os.path.exists(autotune_tmp_file):
```

Comment:
For corner case synchronization issue, should we also check that the file isn't empty?

Reply:
Is there a problem with using an empty autotune config file?
I assume if the main process generates an empty autotune data then we need to use them explicitly in other processes. Otherwise we might end up in a situation when another process decided to actually use some different autotune settings.

Reply:
That is a valid point. If the first process really trigger an empty autotuner, it should mean that there isn't any autotuner call in that graph. That shouldn't be an error.